### ISSUE TITLE
EarlyStopping add initial warm-up field to allow initial convergence step (KL_loss in VAEs)

### ISSUE THREAD

**Issue by inonbe** [keras-team/keras#16793]
If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.

TensorFlow version (2.5.0):
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

I'm training a VAE using tf.keras API and using the EarlyStopping callback.
During the initial training phase there are few epochs that are considered initial convergence
and i don't want them to count for in the patience timeout counter (attached reference as this is in the context 
of the solution for KL Vanishing in VAEs) [https://arxiv.org/abs/1903.10145](url)
i.e. i would like to add for example 20 initial epochs in which the KL_Loss in the VAE
will be disregarded and therefore the EarlyStopping patience wait time-out for improvement should
not yet step in. After this initial warm of few epochs everything in terms of this early stopping callback
should behave as expected.

**Will this change the current api? How?**
yes, i suggest to add one more field to the EarlyStopping callback which will be
called "init_patience" (defaults to 0) and in cases the user calls the callback with values
larger to 0 it will used to halt the self.wait increment until the number of 
epochs are larger than this new API value. 

**Who will benefit from this feature?**
Anyone or in any case where you want to use this call back but 
the loss is initially not diminishing because of initial different behaviour in the loss function
or label weights etc.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):
I would suggest to add as i mentioned above new api "init_patience" and
maintain new member with the same name self.init_patience.

In the "EarlyStopping" callback's function "on_epoch_end" - add to the return condition 
another sub condition so that it returns until epoch is above the init_patiance.

        def on_epoch_end(self, epoch, logs=None):   
              current = self.get_monitor_value(logs)   
              if current is None or epoch <= self.init_patience:     
                 return

**Comment by qlzh727** [comment]
Assign to Rick and Divya for more inputs.

**Comment by divyashreepathihalli** [comment]
@inonbe Thank you for this feature request. It sounds like a great addition to the call back. We can use an intuitive name like `start_from`. Please go ahead and submit the PR and our team will review it.

**Comment by inonbe** [comment]
thanks @divyashreepathihalli - I'm submitting the PR right away.

**Comment by divyashreepathihalli** [comment]
PR merged! so closing this.
